### PR TITLE
Implement everywhere_powerplants

### PR DESCRIPTION
**Pull request by euronion**
## Changes proposed in this Pull Request

The combination of  `conventional_carriers` and `extendable_carriers` currently only allows for expanding conventional capacities in locations, where historically powerplants of that carrier where located.
This PR adds a new configuration option to place conventional powerplants at every network node, irrespective whether powerplants have historically existed there.

## Checklist

- [x] I tested my contribution locally and it seems to work fine.
- [x] Code and workflow changes are sufficiently documented.
- [n/a] Changed dependencies are added to `envs/environment.yaml`.
- [x] Changes in configuration options are added in all of `config.default.yaml`.
- [x] Changes in configuration options are also documented in `doc/configtables/*.csv`.
- [x] A release note `doc/release_notes.rst` is added.

### CODE REVIEW

**Comment by euronion** [comment]
Not sure why the `environment.fixed.yaml` got reformatted automatically by pre-commit

**Review by fneum — APPROVED** [review]
LGTM, nicely documented :)

File: scripts/build_powerplants.py [hunk 1]
```diff
@@ -97,6 +108,40 @@ def add_custom_powerplants(ppl, custom_powerplants, custom_ppl_query=False):
     )
 
 
+def add_everywhere_powerplants(ppl, substations, everywhere_powerplants):
```
Comment (fneum) [inline scripts/build_powerplants.py:111]: Shouldn't this be `everywhere_powerplants` be `everywhere_ppl`

Comment (fneum) [reply scripts/build_powerplants.py:111]: Ah, you were quicker.

**Comment by euronion** [comment]
These errors are what I get for changing the variable names last minute ._.

**Comment by euronion** [comment]
Btw.: Reformatting of `environment.fixed.yaml` and missing REUSE compliance of the file related bf5c323 because it was pushed directly. Fixing it here.

**Comment by fneum** [comment]
Hmm, gibt noch einige Probleme:

```
 Traceback (most recent call last):
  File "/home/runner/work/pypsa-eur/pypsa-eur/.snakemake/scripts/tmp9uspklvi.add_existing_baseyear.py", line 678, in <module>
    add_power_capacities_installed_before_baseyear(
  File "/home/runner/work/pypsa-eur/pypsa-eur/.snakemake/scripts/tmp9uspklvi.add_existing_baseyear.py", line 199, in add_power_capacities_installed_before_baseyear
    df_agg["grouping_year"] = np.take(
                              ^^^^^^^^
  File "/home/runner/micromamba/envs/pypsa-eur/lib/python3.11/site-packages/numpy/core/fromnumeric.py", line 192, in take
    return _wrapfunc(a, 'take', indices, axis=axis, out=out, mode=mode)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/micromamba/envs/pypsa-eur/lib/python3.11/site-packages/numpy/core/fromnumeric.py", line 56, in _wrapfunc
    return _wrapit(obj, method, *args, **kwds)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/micromamba/envs/pypsa-eur/lib/python3.11/site-packages/numpy/core/fromnumeric.py", line 45, in _wrapit
    result = getattr(asarray(obj), method)(*args, **kwds)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
IndexError: index 11 is out of bounds for axis 0 with size 11
```

**Comment by euronion** [comment]
Thanks! :)

**Pull request merged**